### DISCUSSION TITLE
Table Scroll

### DISCUSSION THREAD

**Discussion by hanan-commits-work** [iTwin/iTwinUI-react#580] [Q&A] [answered]
Greetings. I am having trouble with table scrolling...
The table is displayed after pressing a button. But then only a limited number of entries are shown and I can see there are other entries in the table, but I cant scroll down. But when i close the table and open it again, everything seems to start working.

this is happening when i open the table the first time,
![image](https://user-images.githubusercontent.com/89520756/157395289-a5489b5e-3c81-4d4a-b467-2d2c7363e68e.png)
and after the second time,
![image](https://user-images.githubusercontent.com/89520756/157395325-e2f2746c-1a83-4ce2-98b6-4d516f7c4c61.png)

Here is my code...
```
//Here, the tState is a global variable for maintaining `tableState`
var tState : TableState<DemoData>;

//Here, the itemProps are the realityModels that are available in my ContextShare
 const [itemProps, setItemProps] = useState<ContextRealityModelProps[]>([]);

  useEffect(() => {
    const realityModels = cRealityData.getRealityModels();
    if (realityModels) {
      setItemProps(realityModels);
    } 
  }, [itemProps]);

//FYI: Please note that this onSelect is called by onMultiSelect in order to maintain the table state
  const onSelect = useCallback(
    (selectedData: DemoData[] | undefined, state?: TableState<DemoData>) => {
     // tableState.current = state;
      if (state)
        tState = state;
    },
    []
  );


 const sampleData = useMemo(() => {
    const size = itemProps.length;
    const arr = new Array(size);
    for (let i = 0; i < size; ++i) {
      arr[i] = {
        RealityModels: itemProps[i].name,
      };
    }
    return arr;
  }, [itemProps]); 



 <Table
        enableVirtualization 
        data={data}
        columns={Columns}
        emptyTableContent="No data."
        isLoading={false}
        styleType="zebra-rows"
        density="extra-condensed"
        isSelectable={true}
        
        onSelect={onMultiSelect}
        initialState={tState}
        style={{maxHeight: 300}}
      />
```

**Comment by veekeys** [comment, marked as answer]
Hello @hanan-commits-work 

Have you tried disabling virtualization?
Would you mind creating a sandbox with minimal code to reproduce this issue?
https://codesandbox.io/s/itwinui-react-example-son74

Currently, it is pretty hard to just guess, what is going on there...

**Reply by hanan-commits-work** [reply]
Oh this worked. 
Can you explain how, because I thought that virtualization was responsible for scrolling.

**Reply by veekeys** [reply]
Thats why I asked for the sandbox, so I could investigate, what is going on there, because now I just guessed.
Could it be that you mount the table component, but just hide it initally?
